### ISSUE TITLE
Sync stage Generate Storage History fails (too many open files). Starts over.

### ISSUE THREAD

**Issue by tjayrush** [ledgerwatch/turbo-geth#1211]
#### System information

TurboGeth version: `latest master`
OS & Version: OSX

#### Expected behaviour

Sync

#### Actual behaviour

Failed sync

#### Steps to reproduce the behaviour

I started the sync about a week ago. It seemed to be going okay, but I didn't watch it very closely. The last two days, when I looked at it, it seems to be restarting the `Generate Storage History` stage, but I couldn't tell. Today, I watched and saw the following message. I've highlighted the important **error message**:

```
058
INFO [10-08|19:36:44.179] ETL [1/2] Extracting                     from=PLAIN-SCS block=10685795 alloc=1.42GiB    sys=3.80GiB   numGC=1058
INFO [10-08|19:37:20.773] Flushed buffer file                      name=/Volumes/Samsung_T4/turbo_geth/etl-temp/tg-sync-sortable-buf771037646 alloc=1.23GiB    sys=3.80GiB   numGC=1065
INFO [10-08|19:37:20.851] ETL [1/2] Extracting                     from=PLAIN-SCS block=10709696 alloc=1.23GiB    sys=3.80GiB   numGC=1065
INFO [10-08|19:37:44.178] ETL [1/2] Extracting                     from=PLAIN-SCS block=10729991 alloc=631.10MiB  sys=3.80GiB   numGC=1070
INFO [10-08|19:37:59.810] Flushed buffer file                      name=/Volumes/Samsung_T4/turbo_geth/etl-temp/tg-sync-sortable-buf583157461 alloc=1.33GiB    sys=3.80GiB   numGC=1072
INFO [10-08|19:38:14.178] ETL [1/2] Extracting                     from=PLAIN-SCS block=10748149 alloc=652.08MiB  sys=3.80GiB   numGC=1074
INFO [10-08|19:38:38.652] Flushed buffer file                      name=/Volumes/Samsung_T4/turbo_geth/etl-temp/tg-sync-sortable-buf771943984 alloc=1.52GiB    sys=3.80GiB   numGC=1078
INFO [10-08|19:38:44.178] ETL [1/2] Extracting                     from=PLAIN-SCS block=10766579 alloc=301.41MiB  sys=3.80GiB   numGC=1080
INFO [10-08|19:39:21.102] etl: temp files removed successfully     total size="46.8 GB"

**
WARN [10-08|19:39:22.974] Synchronisation failed, retrying         err="storage history index: fail to generate index: open /Volumes/Samsung_T4/turbo_geth/etl-temp/tg-sync-sortable-buf002894799: too many open files"
WARN [10-08|19:39:27.548] Multiple headers for single request      peer=59c94b9d4eb85f33 headers=0
WARN [10-08|19:39:27.548] Synchronisation failed, dropping peer    peer=59c94b9d4eb85f33 err="action from bad peer ignored: multiple headers (0) for single request"
**

INFO [10-08|19:39:28.071] Sync stage 1/13. Download headers...
spawn sync 2
INFO [10-08|19:39:28.911] Imported new block headers               count=384 elapsed=494.616ms number=11017755 hash="6738fa…352bc2" age=23m12s
```

After that message I killed the process (sorry no stack trace), and restarted the sync. Everything went well until the `Generate Storage History`, which had previously appeared to have finished, started over at block `3444312`.

The message `too many open files` is of interest.

I'm not sure, but it feels like this happened more than once (as I said, it felt like it did, but I wasn't previously watching closely enough).

**Comment by AlexeyAkhunov** [comment]
Maybe you can bring back these files (I removed them some time ago) back to fix this: https://github.com/ethereum/go-ethereum/tree/master/common/fdlimit

**Comment by tjayrush** [comment]
Problem may be in this commit: https://github.com/ledgerwatch/turbo-geth/commit/961b1d57c4efc1cf658ce73fd4050238ea11c767. I'll put this code back and see if that fixes the problem.

**Comment by tjayrush** [comment]
I added back code to the increase in file descriptors as mentioned above, but before I did I ran the sync which watching how many open files the process had (`lsof -p <PID>`). The number of open files definitely went over 256 (which is the default max on my machine apparently).

Prior to the change, the problem was easily repeatable (ignoring the fact it took a long time) and the sync would not proceed past Generate Storage History (stage 9 of 13).

After the change, it works.

PR incoming...
